assembly/src/parsers/io_ops/mem_ops.rs

```diff
@@ -164,341 +138,385 @@ fn push_mem_addr(span_ops: &mut Vec<Operation>, op: &Token) -> Result<(), Assemb
             expected
         );
 
         // extra value provided to mem variant
-        let op_str = format!("{}.mem.0.1", base_op);
+        let op_str = format!("{}.0.1", base_op);
         let op_extra_val = Token::new(&op_str, pos);
         let expected = AssemblyError::extra_param(&op_extra_val);
         assert_eq!(
             get_parsing_error(base_op, &op_extra_val, num_proc_locals),
             expected
         );
     }
+
+    /// Test that an instruction for a local memory operation is properly formed. It can be used to
+    /// test parameter inputs for loc_load, loc_store, loc_loadw, and loc_storew.
+    fn test_parse_local_invalid(base_op: &str) {
+        let num_proc_locals = 1;
+
+        // fails when immediate values to a {loc_load|loc_store|loc_loadw|loc_storew}.i operation are
+        // invalid or missing
+        let pos = 0;
+
+        // insufficient values provided
+        let op_val_missing = Token::new(base_op, pos);
+        let expected = AssemblyError::missing_param(&op_val_missing);
+        assert_eq!(
+            get_parsing_error(base_op, &op_val_missing, num_proc_locals),
+            expected
+        );
+
+        // invalid value provided to local variant
+        let op_str = format!("{}.abc", base_op);
+        let op_val_invalid = Token::new(&op_str, pos);
+        let expected = AssemblyError::invalid_param(&op_val_invalid, 1);
+        assert_eq!(
+            get_parsing_error(base_op, &op_val_invalid, num_proc_locals),
+            expected
+        );
+
+        // no procedure locals declared
+        let op_str = format!("{}.{}", base_op, 1);
+        let op_no_locals = Token::new(&op_str, pos);
+        let expected = AssemblyError::invalid_op_with_reason(
+            &op_no_locals,
+            "no procedure locals were declared",
+        );
+        assert_eq!(get_parsing_error(base_op, &op_no_locals, 0), expected);
+
+        // provided local index is outside of the declared bounds of the procedure locals
+        let op_str = format!("{}.{}", base_op, num_proc_locals);
+        let op_val_invalid = Token::new(&op_str, pos);
+        let expected = AssemblyError::invalid_param_with_reason(
+            &op_val_invalid,
+            1,
+            format!(
+                "parameter value must be greater than or equal to 0 and less than or equal to {}",
+                num_proc_locals - 1
+            )
+            .as_str(),
+        );
+        assert_eq!(
+            get_parsing_error(base_op, &op_val_invalid, num_proc_locals),
+            expected
+        );
+
+        // extra value provided to local variant
+        let op_str = format!("{}.0.1", base_op);
+        let op_extra_val = Token::new(&op_str, pos);
+        let expected = AssemblyError::extra_param(&op_extra_val);
+        assert_eq!(
+            get_parsing_error(base_op, &op_extra_val, num_proc_locals),
+            expected
+        );
+    }
+
+    fn test_parse_local(
```

Comment:
This looks great! Well done! One nit: let's add a simple doc comment to this function.